presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -383,32 +356,13 @@ private static String objectToString(Object object, CassandraType elemType)
             case FLOAT:
             case DECIMAL:
                 return object.toString();
+            case LIST:
+            case SET:
+                return buildArrayValue((Collection<?>) object, dataType.getTypeArguments().get(0));
+            case MAP:
+                return buildMapValue((Map<?, ?>) object, dataType.getTypeArguments().get(0), dataType.getTypeArguments().get(1));
             default:
-                throw new IllegalStateException("Handling of type " + elemType + " is not implemented");
-        }
-    }
-
-    @Override
-    public CassandraType getCassandraType()
-    {
-        if (getTypeArgumentSize() == 0) {
-            return this;
-        }
-        else {
-            // must not be called for types with type arguments
-            throw new IllegalStateException();
-        }
-    }
-
-    @Override
-    public List<CassandraType> getTypeArguments()
-    {
-        if (getTypeArgumentSize() == 0) {
-            return null;
-        }
-        else {
-            // must not be called for types with type arguments
-            throw new IllegalStateException();
+                throw new IllegalStateException("Handling of type " + cassandraType + " is not implemented");
```

Comment:
```suggestion
                throw new IllegalStateException("Unsupported type: " + cassandraType);
```

(otherwise you'd use `String#format` here)